### PR TITLE
[ci] avoid migration failure due to new empty lines in structure.sql

### PR DESCRIPTION
**Pull request by adrianschroeter**
note, this is untested but we fail since 10 days now, so it can't be worse ;)

### CODE REVIEW

**Comment by M0ses** [comment]
LGTM

**Comment by adrianschroeter** [comment]
Urgs ... 4267c5c0127427e8f2fecb1fe4398369114e3dc3 made the entire test obsolete :/

**Comment by adrianschroeter** [comment]
@evanrolfe we do test now that the current code can create a database with old data and migrate it.

The former code was testing that the database created by 2.5 code base can get migrated, which is the real life scenario IMHO.

**Comment by bgeuken** [comment]
> adrianschroeter commented 33 minutes ago • edited
> Urgs ... 4267c5c made the entire test a void operation :/

Though the whole test is somewhat of a void operation in itself:>

**Comment by bgeuken** [comment]
LGTM

**Comment by evanrolfe** [comment]
@adrianschroeter which migraiton is failing when migrating from 2.5 db?

Also keep in mind that the 2.5 database is whats contained in the first migration:
https://github.com/openSUSE/open-build-service/blob/master/src/api/db/migrate/001_initial_database.rb

**Comment by adrianschroeter** [comment]
synced the database setup in first migration with 2.5 database.

using the database snapshot for migration test again. zero diff on my laptop now, but I didn't test with package build.

**Comment by adrianschroeter** [comment]
adding also old migrations and move the block of adding constraints before adding data to be able to see which data conflicts.

**Comment by evanrolfe** [comment]
@adrianschroeter i'd still like to know what i asked earlier:

> please give us more details. Exactly what line/command is failing and what is the error message?

File: src/api/db/migrate/001_initial_database.rb [hunk 1]
```diff
@@ -915,6 +855,79 @@ def self.up
...
...0610100244),(20130612151549),(20130618083665),(20130621083665),(20130626160000),(20130627193722),(20130702083665),(20130702203665),(20130723055536),(20130725123636),(20130726144516),(20130802183104),(20130802183717),(20130802190951),(20130805073101),(20130807071147),(20130814071147),(20130816183104),(20130817082602),(20130819114303),(20130820151442),(20130830043205),(20130903114302),(20130904071147),(20130910162318),(20130917124132),(20130920090004),(20130930130128),(20131005225515),(20131006000000),(20131006162847),(20131020151037),(20131020165316),(20131021063641),(20131022114302),(20131023063641),(20131027122410),(20131028085325),(20131029112259),(20131105112259),(20131106112233),(20131111090310),(20131111193512),(20131111194720),(20131112140033),(20131120193512),(20131123113417),(20131124071042),(20131125071042),(20131125101042),(20131126074753),(20131126205430),(20131127091624),(20131209095749),(20131209103450),(20131210182719),(20131218071042),(20140113110551),(20140123063641),(20140123071042),(20140124071042),(21),(22),(23),(24),(25),(26),(27),(28),(29),(3),(30),(31),(32),(33),(34),(35),(36),(37),(38),(39),(4),(40),(41),(42),(43),(44),(45),(46),(47),(48),(5),(6),(7),(8),(9);
```
Comment (evanrolfe) [inline src/api/db/migrate/001_initial_database.rb]: None of these migrations exist in the codebase?

**Comment by adrianschroeter** [comment]
The initial failure is in the build logs of the obs-server package, different number of empty lines due to some commit.

When using the DB lots of smaller changes, mostly ordering issues (you can see this as part of commit be3aeb3 ) and some different options for tables and columns.

File: src/api/db/migrate/001_initial_database.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/db/migrate/001_initial_database.rb]: they existed in former releases (also when you do a fresh installation of 2.5)

Comment (hennevogel) [inline src/api/db/migrate/001_initial_database.rb]: How can this be?

File: src/api/db/migrate/001_initial_database.rb [hunk 2]
```diff
@@ -40,8 +40,8 @@ def self.up
       t.integer "group_id"
       t.index ["attrib_namespace_id", "user_id", "group_id"], name: "attrib_namespace_user_role_all_index", unique: true, using: :btree
       t.index ["attrib_namespace_id"], name: "index_attrib_namespace_modifiable_bies_on_attrib_namespace_id", using: :btree
-      t.index ["group_id"], name: "bs_group_id", using: :btree
       t.index ["user_id"], name: "bs_user_id", using: :btree
+      t.index ["group_id"], name: "bs_group_id", using: :btree
```
Comment (hennevogel) [inline src/api/db/migrate/001_initial_database.rb]: Can you please recreate this without all these whitespace and line changes?

File: src/api/db/migrate/001_initial_database.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/db/migrate/001_initial_database.rb]: they got removed meanwhile

Comment (evanrolfe) [inline src/api/db/migrate/001_initial_database.rb]: Why do we need them in schema_migrations then?

**Comment by adrianschroeter** [comment]
JFYI, Evan want to have a look at it again. We will merge tomorrow morning if no new issue pops up.

File: src/api/db/migrate/001_initial_database.rb (hunk 1 above)
Comment (adrianschroeter) [inline src/api/db/migrate/001_initial_database.rb]: (as discussed on phone), we can drop them, but we need to adapt the test to ignore the missing entries then.

**Comment by evanrolfe** [comment]
@adrianschroeter I've tried to recreate the migration from 2.5 error on the vagrant dev machine but I dont get any failed migrations:
```
vagrant ssh
cd /vagrant/src/api
xzcat test/dump_2.5.sql.xz > test/dump_2.5.sql
# Remove line #21 "USE api_25;" from the dump because it doesnt apply here
rake db:drop db:create
mysql -h localhost -u root -p api_development < test/dump_2.5.sql
rake db:migrate
```

After doing that, this is the diff I get from structure.sql between what it is locally and what it is in master:
https://gist.github.com/evanrolfe/b8f093f6673d03e6e7e4bbf364877698

As you can see the only differences are that "USING BTREE" has been removed (which we dont need as its default) and that the schema_migrations are included (which we agreed that we also dont need).

So it seems that the same script running on vagrant versus running on the build machine is producing different results. We should investigate this..

However, if I do `git checkout adrian/master; rake db:drop db:create db:migrate` then there is no diff in structure.sql so I'm ok for this to be merged as long as the non-existant migrations are removed from the "INSERT INTO `schema_migrations`" statement in in `db/migrate/001_initial_database.rb`.

Also it really seems like we need to move this test into circleCI so we catch any bugs like this before we try to build the packages..

**Comment by adrianschroeter** [comment]
okay, merged for now to see if packages are building on all platforms now or if we differ there as well.

Let's follow up afterwards on how to proceed here.

**Comment by adrianschroeter** [comment]
at least the migration tests did succeed now on all platforms. So it seems to be more an issue within vagrant vs. other installations

**Comment by evanrolfe** [comment]
@adrianschroeter can you please remove the non-existitant migrations from the 001 migration as I did say that was a condition for merging this:

> I'm ok for this to be merged as long as the non-existant migrations are removed from the "INSERT INTO schema_migrations" statement

**Comment by hennevogel** [comment]
So we merge now despite failing tests, despite people leaving comments requesting changes and we also dismiss reviews

👏 👏 👏 what a load of 🐄 💩

**Comment by adrianschroeter** [comment]
@hennevogel huh? Only Evan wanted changes and I came to an agreement with him. And the merge did not introduce new failures, it just uncovered failures from testsuite which did not run since quite some time because this failure.

**Comment by hennevogel** [comment]
@adrianschroeter I have also requested changes. Changes you did not do. Instead of waiting for me to okay them (which I wouldn't have) you have dismissed my review and merged...

**Comment by adrianschroeter** [comment]
I dismissed it with "done" comment... because I did change the empty line thingy.

**Comment by evanrolfe** [comment]
@adrianschroeter I did say that the one condition for merging was to delete the non-existent migration rows from schema_migrations table in `db/migrate/001_initial_database.rb` which still has not been done..

**Comment by adrianschroeter** [comment]
@evanrolfe right, fine with me, just thought that we agreed that we can do this as second step to unblock the package building... will have a look into that next week.

**Comment by evanrolfe** [comment]
@adrianschroeter perhaps that was a misunderstanding, I was expecting that to be done before the merge or at the very least immediately after.

**Pull request merged**